### DISCUSSION TITLE
amazon.com: ads

### DISCUSSION THREAD

**Discussion by Elliander** [uBlockOrigin/uAssets#21612] [Q&A]
### Prerequisites

- [X] This is **NOT** a [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL address of the web page

`https://www.amazon.com/`

### Category

ads

### Description

On multiple websites ads keep reappearing after I block them. For example, Amazon has an annoying full screen video ad for sports that I'd rather not have to see. I used "Element Picker Mode" to remove the ad and it gets replaced with a black box, but on page refresh the ads return. If I keep going eventually I will get them all, but the next day the ads return. I have observed the same behavior with countless websites. I think they are rotating the file names to bypass adblockers. If it were possible to block whatever element appears on a certain area of a page regardless of it's name though I think I'd be able to block these kinds of ads.

For example, today, this is the element that I blocked on Amazon which resulted in the ads finally going away:

`##._desktop-tall-hero-video_style_lazy-video-wrapper__WM56t
`
Tomorrow, however, I expect that name to be slightly different. 

### Other extensions used

I used to also use AdBlock Plus, but uninstalled that extension after they started pushing popups of their own.

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

I don't think showing the ads would really help anything, but if it really is necessary I'll edit to put screenshots of the ads later.

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.54.0
Chromium: 92
filterset (summary):
 network: 106932
 cosmetic: 44177
 scriptlet: 19738
 html: 0
listset (total-discarded, last-updated):
 removed:
  user-filters: null
 default:
  easylist: 76133-16, now
  easyprivacy: 33136-65, now
  plowe-0: 3782-1151, now
  ublock-badware: 7653-138, now
  ublock-filters: 36467-345, now
  ublock-privacy: 952-12, now
  ublock-quick-fixes: 103-5, now
  ublock-unbreak: 2176-32, now
  urlhaus-1: 12355-0, now
filterset (user): [array of 92 redacted]
trustedset:
 added: [array of 1 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 3117 ms
 maxAssetCacheWait: 1727 ms
popupPanel:
 blocked: 67
 network:
  amazon.com: 65
  amazon-adsystem.com: 1
  ssl-images-amazon.com: 1
 extended:
  ###nav-swmslot
  ##div[cel_widget_id^="adplacements:"]
  ##.ad-center
  ##+js(json-prune, cuepointPlaylist)
  ##+js(xml-prune, xpath(//*[name()="Period"][.//*[@value="Ad"]] |…
```
</details>

**Comment by mapx-** [comment]
> they started pushing popups of their own

Are you sure it's about abp popups and not some malware in your system ?

**Comment by gorhill** [comment]
> ads keep reappearing after I block them

>  removed:
>   user-filters: null

You disabled your own list of filters, so whatever filters you create won't be applied on page reload. Re-enable _"My filters"_ in _"Filter lists"_ pane.

**Comment by Elliander** [comment]
> > ads keep reappearing after I block them
> 
> > removed:
> > user-filters: null
> 
> You disabled your own list of filters, so whatever filters you create won't be applied on page reload. Re-enable _"My filters"_ in _"Filter lists"_ pane.

I don't understand why a fresh install would default to disabling my own filters immediately after creating them (or how filters can briefly work while supposedly disabled), but ok, I followed those directions and it's still a problem.

For example, one video add has this code when I select it with the object picker:
`
##.pd_rd_r-5f1cd3b8-8ae7-4bd0-aab3-f9ba4b7f4187.pd_rd_w-4XopZ.pd_rd_wg-jzsPJ.celwidget.desktop-gateway-atf_7c2a8e11-21a0-46b5-a65e-d58b72a4d484.csm-placement-id-5af45aa7-715b-4ed8-811b-7f9befef5ba2.gw-col
`
But on page refresh the ad returns and thhis is the code:

`##.pd_rd_r-5f8c5d7d-1445-4897-920d-f63da085a01f.pd_rd_w-0Lyr1.pd_rd_wg-WKIZh.celwidget.desktop-gateway-atf_7c2a8e11-21a0-46b5-a65e-d58b72a4d484.csm-placement-id-5af45aa7-715b-4ed8-811b-7f9befef5ba2.gw-col`

As you can see, it's the same ad, but the text is ever so slightly varied to get through the adblocker. Are modern adblockers so easy to defeat that a slight change like that is enough to let ads through?

Oh, and I wasn't aware I was required to reply every single day or the case gets closed. That's not cool. Especially on a Holiday week. 

I guess I am going to have to just reopen the case every other week? That is SO beyond stupid....

**Comment by gorhill** [comment]
> I don't understand why a fresh install would default to disabling my own filters

It doesn't, they are enabled by default, and only you can disable it by un-checking the list.

**Comment by gorhill** [comment]
What you are asking here is a support question, and with the information you provide, we don't know what you are trying to remove from where -- that's why we are asking for screenshots to help save time.

Converting to discussion because you are not reporting something to be added to filter lists.

**Comment by gorhill** [comment]
You provided a screenshot in [the duplicate issue](https://github.com/uBlockOrigin/uAssets/issues/21611):

<details><summary>screenshot</summary>

![amazon_home_page](https://github.com/uBlockOrigin/uAssets/assets/15022396/2982abf1-4b98-46af-b6e5-a06787f8e116)
</details>

> Amazon has an annoying full screen video ad for sports that I'd rather not have to see

I could remove it after using the element picker a second time but selecting the second choice in "Network filters" section: the picker remembers previous choices and may suggest a filter with wildcards to create a filter which will match the previous and current choice. 

For that specific case however, it's best to use the suggested cosmetic filter `###desktop-banner`:

![Screenshot from 2023-12-27 20-29-43](https://github.com/uBlockOrigin/uAssets/assets/585534/f108f8bd-bc87-4ea9-a2fd-48ccd2c38622)

So the filter to accomplish what you want is `amazon.com###desktop-banner`.

The element picker is a tool to assist you in creating filters, but only you know what you want to filter, the element picker can't read your mind. It offers you a list of suggested filters according to where you clicked, and you have to pick the one which does what you want ([demo](https://www.youtube.com/watch?v=8TvCGWwQr5o), [documentation](https://github.com/gorhill/uBlock/wiki/Element-picker)). You can also adjust the result using the sliders or manually modify the filter in the text area.

> I didn't have these problems when I was using Adblock plus

It has the same problem, but worse since it doesn't offer you a list of alternative filters to choose from:

![image](https://github.com/uBlockOrigin/uAssets/assets/585534/9913146d-4590-4d53-b0c5-dabbf16b117f)

Notice that with uBO's element picker, it also suggests you a network filter with wildcard in it, as will try to replace the changing parts with wildcards after it merges with previously generated network filters:

![Screenshot from 2023-12-27 20-35-29](https://github.com/uBlockOrigin/uAssets/assets/585534/ef625e33-f886-4655-a1df-efd8b90546d4)

Adblock Plus's picker is very bare, and certainly does not offer that level of assistance found in uBO's picker, so I don't know which part of ABP's picker you think works better, it doesn't, and you are stuck with only one suggested filter with it, the same one which caused you issue here. The solution with uBO was to pick other suggested filters.
